examples/DocExampleApis/Program.cs

```diff
@@ -122,4 +137,134 @@ public static async Task Example_API_Delete(CacheClient cacheClient)
             throw new Exception($"An error occurred while attempting to delete key 'test-key' from cache 'test-cache': {error.ErrorCode}: {error}");
         }
     }
+
+    public static async Task Example_API_GenerateDisposableToken(IAuthClient authClient)
+    {
+        // Generate a disposable token with read-write access to a specific key in one cache
+        var oneKeyOneCacheToken = await authClient.GenerateDisposableTokenAsync(
+            DisposableTokenScopes.CacheKeyReadWrite("squirrels", "mo"),
+            ExpiresIn.Minutes(30)
+        );
+        if (oneKeyOneCacheToken is GenerateDisposableTokenResponse.Success token1)
+        {
+            
+            // logging only a substring of the tokens, because logging security credentials is not advisable :)
+            Console.WriteLine("The generated disposable token starts with: " + token1.AuthToken.Substring(0, 10));
+            Console.WriteLine("The token expires at (epoch timestamp): " + token1.ExpiresAt.Epoch());
+        }
+        else if (oneKeyOneCacheToken is GenerateDisposableTokenResponse.Error err)
+        {
+            Console.WriteLine("Error generating disposable token: " + err.Message);
+        }
+
+        // Generate a disposable token with read-write access to a specific key prefix in all caches
+        var keyPrefixAllCachesToken = await authClient.GenerateDisposableTokenAsync(
+            DisposableTokenScopes.CacheKeyPrefixReadWrite(CacheSelector.AllCaches, "squirrel"),
+            ExpiresIn.Minutes(30)
+        );
+        if (keyPrefixAllCachesToken is GenerateDisposableTokenResponse.Success token2)
+        {
+            // logging only a substring of the tokens, because logging security credentials is not advisable :)
+            Console.WriteLine("The generated disposable token starts with: " + token2.AuthToken.Substring(0, 10));
+            Console.WriteLine("The token expires at (epoch timestamp): " + token2.ExpiresAt.Epoch());
+        }
+        else if (keyPrefixAllCachesToken is GenerateDisposableTokenResponse.Error err)
+        {
+            Console.WriteLine("Error generating disposable token: " + err.Message);
+        }
+
+        // Generate a disposable token with read-only access to all topics in one cache
+        var allTopicsOneCacheToken = await authClient.GenerateDisposableTokenAsync(
+            DisposableTokenScopes.TopicSubscribeOnly(CacheSelector.ByName("squirrel"), TopicSelector.AllTopics),

```

Comment:
This one can be changed to remove `CacheSelector.ByName` once the additional predefined scope functions are released

Reply:
Ah, so we're missing an overload that takes `(string, TopicSelector)`. Good catch. Would you mind making a ticket to remind us to audit these functions and add signatures for `(string, XYZSelector)` please?

Reply:
Thank you for adding those!